### DISCUSSION TITLE
EEPROM support for FT24C256A BBT SKR 1.4

### DISCUSSION THREAD

**Discussion by K2AAE** [terjeio/grblHAL#303] [General]
The SKR 1.4 EEPROM Module has a FT24C256A device, looks to be pin compatible with 24LC16B/24AAxxx devices allowing a device swamp to be compatible with the existing drivers, it appears. Has anyone tried to swamp out the part?

**Comment by terjeio** [comment]
>  Has anyone tried to swamp out the part?

It should not be neccesary to do that, try with setting EEPROM_ENABLE  to 2 for [64 byte pagesize](https://datasheet.lcsc.com/lcsc/1912111437_FMD-Fremont-Micro-Devices-FT24C256A-ESR-B_C417372.pdf) in _my_machine.h_ first:

https://github.com/terjeio/grblHAL/blob/9f405bae810fa42c60ebaf2c21c2ff553e46a049/drivers/LPC1769/src/my_machine.h#L39

You have to change the board map as well as it does not allow EEPROM, looks like replacing this code

https://github.com/terjeio/grblHAL/blob/9f405bae810fa42c60ebaf2c21c2ff553e46a049/drivers/LPC1769/src/btt_skr_1.4_turbo_map.h#L33-L35

with 

`#define I2C_ENABLE 2`

should work.
